### PR TITLE
Jira connector

### PR DESCRIPTION
**Pull request by Shrikey**
DATA-962 - Create JIRA script to fetch tickets per project via the JIRA API

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,7 @@
 .vscode/
 wherescape.log
 .DS_Store
-*.pyc
\ No newline at end of file
+*.pyc
+__pycache__
+virtual/
+config.py
```
Comment (BFJonk) [inline .gitignore:8]: I would have expected instructions on how to use this file in https://github.com/wearespindle/python-wherescape ?

File: wherescape/connectors/jira/readme.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Jira connector for WhereScape. Takes care of reading Projects and issues from
+a Jira environment.
+
+# Preparation
+
+## WhereScape Parameters
+
+Add the following parameters to WhereScape:
+* `jira_user`
```
Comment (BFJonk) [inline wherescape/connectors/jira/readme.md:11]: If you would have use Extended Properties you would have been able to use the Wheredscape Masking function. 

Even more advanced would be to use a JIRA connection Type with the UI configuration.

File: wherescape/connectors/jira/readme.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Host script
+Create a new python host script and add it to the load table. Example code:
+
+```
+import logging
+from wherescape.connectors.jira.python_jira_load_data import jira_load_data_project
+
+if __name__ == "__main__":
+    jira_load_data_project()
+```
```
Comment (BFJonk) [inline wherescape/connectors/jira/readme.md:28]: Where has the try / error Handling gone?

File: wherescape/connectors/jira/readme.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+Host scripts to create:
```
Comment (BFJonk) [inline wherescape/connectors/jira/readme.md:30]: That are lot of host scripts, we could create a script for this?

File: wherescape/connectors/jira/readme.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+* python_jira_create_metadata_project
+* python_jira_create_metadata_issue
+* python_jira_load_data_project
+* python_jira_load_data_issue_full
+* python_jira_load_data_issue_incremental
```
Comment (BFJonk) [inline wherescape/connectors/jira/readme.md:35]: I would have made full / incremental a parameter (?)

File: wherescape/connectors/jira/readme.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+# Usage
+
+First attach the metadata host script to the load table. After creating the
+table, attach the load_data host script to the load table. For issues, after
+the initial load, use the incremental load for better performance.
```
Comment (BFJonk) [inline wherescape/connectors/jira/readme.md:41]: We could even create a small video & upload it to Youtube or Panopto

File: requirements.txt [hunk 7]
```diff
@@ -1,3 +1,5 @@
 notion-client
+pandas
 pyodbc
```
Comment (BFJonk) [inline requirements.txt:3]: maybe for the Readme.md: We of course Assume that (a) Python was installed on the wehrescape server (b) The Wherescape Python package is installed

**Review by BFJonk — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Shrikey) [reply .gitignore:8]: This file is actually not used in the repo. Might remove it from the repo.

File: wherescape/connectors/jira/readme.md (hunk 2 above)
Comment (Shrikey) [reply wherescape/connectors/jira/readme.md:11]: Yes, indeed, the JIRA connection type should be the future. Will make a ticket for this.

File: wherescape/connectors/jira/readme.md (hunk 3 above)
Comment (Shrikey) [reply wherescape/connectors/jira/readme.md:28]: Not necessary anymore with the new logging.

File: wherescape/connectors/jira/readme.md (hunk 4 above)
Comment (Shrikey) [reply wherescape/connectors/jira/readme.md:30]: It's only for 1 time, so I don't see the trouble.

File: wherescape/connectors/jira/readme.md (hunk 6 above)
Comment (Shrikey) [reply wherescape/connectors/jira/readme.md:41]: Yeh, interesting idea.

File: requirements.txt (hunk 7 above)
Comment (Shrikey) [reply requirements.txt:3]: I think that should be obvious already.

File: wherescape/connectors/jira/readme.md (hunk 5 above)
Comment (Shrikey) [reply wherescape/connectors/jira/readme.md:35]: Might be a future extension.

File: wherescape/connectors/jira/python_jira_load_data.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    logging.info(
+        "Start time: %s for jira_load_data" % start_time.strftime("%Y-%m-%d %H:%M:%S")
+    )
+
+    # Initialise WhereScape and get the relevant WhereScape values.
+    logging.info("Connecting to WhereScape")
+    user = wherescape_instance.read_parameter("jira_user")
+    apikey = wherescape_instance.read_parameter("jira_apikey")
+    wherescape_object_id = wherescape_instance.object_key
+    base_url = wherescape_instance.base_uri
```
Comment (BFJonk) [inline wherescape/connectors/jira/python_jira_load_data.py:52]: suggest to rename

File: wherescape/connectors/jira/python_jira_load_data.py [hunk 9]
```diff
@@ -0,0 +1,99 @@
+from datetime import datetime
+import logging
+
+from .jira_wrapper import Jira
+
+from wherescape import WhereScape
```
Comment (BFJonk) [inline wherescape/connectors/jira/python_jira_load_data.py:6]: why is one called wrapper? and the other not? should be consistent?

File: wherescape/logging.py [hunk 10]
```diff
@@ -75,20 +78,37 @@ def flush(self):
...
+
 def initialise_wherescape_logging(wherescape):
     """
     Function to configure the logging for WhereScape.
     """
 
-    logger = logging.getLogger()
-    logger.setLevel(logging.INFO)
     fmt = "[%(levelname)s] %(asctime)s %(filename)s %(funcName)s():%(lineno)i: %(message)s"
     message_format = logging.Formatter(fmt=fmt, datefmt="%H:%M:%S")
```
Comment (BFJonk) [inline wherescape/logging.py:99]: `datefmt` has no date, only hours?

File: wherescape/connectors/jira/__init__.py [hunk 11]
```diff
@@ -0,0 +1,6 @@
+"""
+Module that takes care of the connection to Notion.
```
Comment (BFJonk) [inline wherescape/connectors/jira/__init__.py:2]: JIRA
```suggestion
Module that takes care of the connection to JIRA.
```

File: wherescape/connectors/jira/python_jira_load_data.py (hunk 9 above)
Comment (Shrikey) [reply wherescape/connectors/jira/python_jira_load_data.py:6]: Its how we named them in the other repository. WhereScape is more like the main class with own functions and the connectors are more wrappers around the API functions.

**Pull request closed without merging**